src/winrpath.cxx

```diff
@@ -0,0 +1,1025 @@
+ *  and, using the structure of the COFF format, identifies and renames each location in which
+ *  a mangled DLL name would be found.
+ * 
+ * \param name the absolute path to a dll to be unmangled
+ */
+bool CoffParser::NormalizeName(std::string &name)
+{
+    // The dll is found with and without an extenion, depending on the context of the location
+    // i.e. in the section data, it can be found with both an extension and extensionless
+    //  whereas in the symbol table or linker member strings, it's always found without an extension
+    std::string name_no_ext = strip(name, ".dll");
+
+    // Iterate through the parsed COFF members
+    for (auto mem: this->coff_.members) {
+        int i = 0;
+        // import member names from spack are of the form "      /n" where n is their place
+        // in the longnames member, other members are "/[/]        "
+        // This allows us to determine if we're looking at an import member, and where the offset is
+        while(i < 16 && mem.header->Name[i] != ' ') {
+            ++i;
+        }
+        std::string name_ref = std::string((char*)mem.header->Name, i);
+        if (!endswith(name_ref, "/")) {
+            // We have an import member
+            // Name is longer than 16 bytes, need to lookup name in longname offset
+            int longname_offset = std::stoi(name_ref.substr(1, std::string::npos));
+            // Longnames member is always the third member if it exists
+            // We know it exists at this point due to the success of the conditional above
+            std::vector<char> long_name;
+            // Reconstruct name from location in longnames member
+            int i;
+            for (i = longname_offset; this->coff_.members[2].member->data[i] != '\0'; ++i)
+                long_name.push_back(this->coff_.members[2].member->data[i]);
+            // Ensure Dll name is the one we're looking to perform the rename for
+            if (!strcmp(name.c_str(), std::string(long_name.begin(), long_name.end()).c_str())) {
+                // If so, unmangle it
+                replace_special_characters((char *)&long_name[0], i-longname_offset);
+                // offset of actual longname member
+                int offset = std::streamoff(this->coff_.members[2].offset);
+                this->coffStream->seek(0);
+                // Seek to longname header
+                this->coffStream->seek(offset);
+                // Seek to offset within longname member for a given import name
+                this->coffStream->seek(sizeof(IMAGE_ARCHIVE_MEMBER_HEADER) + longname_offset, std::ios_base::cur);
+                this->coffStream->write(long_name.data(), long_name.size());
+            }
+            // Import member name has been renamed
+            // Now we rename the other DLL references
+            // Import members have two forms, long and short, check for short
+            if(mem.member->is_short) {
```

Comment:
(question) do the tests cover this case?

Reply:
Discussion summary:
As of right now the binaries generated locally do, but we don't have a clear guarantee as to when and why they're created, so we need to investigate this further. Reaching out to experts w/ Windows linkers/PE and looking int LLVM source for Windows import libraries.